libp2p/NodeTable.cpp

```diff
@@ -20,44 +20,68 @@ constexpr unsigned c_handleTimeoutsIntervalMs = 5000;
 
 }  // namespace
 
-constexpr std::chrono::seconds DiscoveryDatagram::c_timeToLive;
-constexpr std::chrono::milliseconds NodeTable::c_reqTimeout;
-constexpr std::chrono::milliseconds NodeTable::c_bucketRefresh;
-constexpr std::chrono::milliseconds NodeTable::c_evictionCheckInterval;
+constexpr chrono::seconds DiscoveryDatagram::c_timeToLive;
+constexpr chrono::milliseconds NodeTable::c_reqTimeout;
+constexpr chrono::milliseconds NodeTable::c_bucketRefresh;
+constexpr chrono::milliseconds NodeTable::c_evictionCheckInterval;
 
-inline bool operator==(
-    std::weak_ptr<NodeEntry> const& _weak, std::shared_ptr<NodeEntry> const& _shared)
+inline bool operator==(weak_ptr<NodeEntry> const& _weak, shared_ptr<NodeEntry> const& _shared)
 {
     return !_weak.owner_before(_shared) && !_shared.owner_before(_weak);
 }
 
 NodeTable::NodeTable(ba::io_service& _io, KeyPair const& _alias, NodeIPEndpoint const& _endpoint,
     bool _enabled, bool _allowLocalDiscovery)
-  : m_hostNodeID(_alias.pub()),
-    m_hostNodeEndpoint(_endpoint),
-    m_secret(_alias.secret()),
-    m_socket(make_shared<NodeSocket>(
-        _io, static_cast<UDPSocketEvents&>(*this), (bi::udp::endpoint)m_hostNodeEndpoint)),
-    m_requestTimeToLive(DiscoveryDatagram::c_timeToLive),
-    m_allowLocalDiscovery(_allowLocalDiscovery),
-    m_timers(_io),
-    m_io(_io)
+  : m_hostNodeID{_alias.pub()},
+    m_hostNodeEndpoint{_endpoint},
+    m_secret{_alias.secret()},
+    m_socket{make_shared<NodeSocket>(
+        _io, static_cast<UDPSocketEvents&>(*this), (bi::udp::endpoint)m_hostNodeEndpoint)},
+    m_requestTimeToLive{DiscoveryDatagram::c_timeToLive},
+    m_allowLocalDiscovery{_allowLocalDiscovery},
+    m_discoveryTimer{make_shared<ba::deadline_timer>(_io)},
+    m_evictionTimer{make_shared<ba::deadline_timer>(_io)},
+    m_io{_io}
 {
     for (unsigned i = 0; i < s_bins; i++)
         m_buckets[i].distance = i;
 
-    if (!_enabled)
+    if (_enabled)
+    {
+        try
+        {
+            m_socket->connect();
+        }
+        catch (exception const& _e)
+        {
+            cwarn << "Exception connecting NodeTable socket: " << _e.what();
+            _enabled = false;
+        }
+    }
+    else cwarn << "\"enabled\" parameter is false";
```

Comment:
You have similar logs twice. Also use quick return pattern as previously.

Reply:
Yea I wanted to avoid redundant logging using the _enabled flag I’ll switch back to early return.